lib/privacy_pass_tokenizer.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+# The TokenChallenge structure contains the type of token, the hostname of the issuer,
+# an optional context to bind to your challenge, and the hostname of your server.
+# iOS 16 and macOS Ventura support token type 2, which uses publicly verifiable RSA Blind Signatures.
+
+# struct {
+#     uint16_t token_type;               // 0x0002, in network-byte order
+#     uint16_t issuer_name_length;       // Issuer name length, in network-byte order
+#     char issuer_name[];                // Hostname of the token issuer
+#     uint8_t redemption_context_length; // Redemption context length (0 or 32)
+#     uint8_t redemption_context[];      // Redemption context, either 0 or 32 bytes
+#     uint16_t origin_info_length;       // Origin info length, in network-byte order
+#     char origin_info[];                // Hostname of your server
+# } TokenChallenge;
+
+class PrivacyPassTokenizer
+  TOKEN_TYPE = 2
+  CHALLENGE_DIGEST_CACHE_PREFIX = "privacy_pass_challenge_digest_"
+
+  # Base64 url encoded
+  def self.issuer_public_key
+    # fastly demo fallback
+    # https://www.fastly.com/blog/private-access-tokens-stepping-into-the-privacy-respecting-captcha-less#demo-configuration
+    # rubocop:disable Layout/LineLength
+    ENV.fetch("PRIVACY_PASS_ISSUER_PUBLIC_KEY",
```

Comment:
Would it make sense to move this also to rubygems.yml/secrets.yml?

Reply:
I can. It's not really a secret but no harm in adding it there.